src/resources/resources.controller.spec.ts

```diff
@@ -1,25 +1,271 @@
+            createdAt: new Date(Date.now()),
+            updatedAt: new Date(Date.now()),
+        },
+    ];
+
+    const mockResourcesService = {
+        createNewResource: jest.fn(),
+        findAllResources: jest.fn(),
+        updateResource: jest.fn(),
+        removeResource: jest.fn(),
+    };
     beforeEach(async () => {
         const module: TestingModule = await Test.createTestingModule({
             controllers: [ResourcesController],
-            providers: [
-                {
-                    provide: ResourcesService,
-                    useValue: {},
-                },
-            ],
-        }).compile();
+            providers: [ResourcesService],
+        })
+            .overrideProvider(ResourcesService)
+            .useValue(mockResourcesService)
+            .compile();
 
         controller = module.get<ResourcesController>(ResourcesController);
     });
 
     it("should be defined", () => {
         expect(controller).toBeDefined();
     });
+
+    describe("createNewResource", () => {
+        it("createNewResource service should be defined", async () => {
+            expect(controller.createNewResource).toBeDefined();
+        });
+
+        it("should create a new resource", async () => {
+            mockResourcesService.createNewResource.mockResolvedValueOnce(
+                mockResource,
+            );
+
+            const result = await controller.createNewResource(
+                requestMock,
+                mockTeamId,
+                dtoCreateMock,
+            );
+
+            expect(result).toEqual(mockResource);
+            expect(mockResourcesService.createNewResource).toHaveBeenCalledWith(
+                requestMock,
+                dtoCreateMock,
+                mockTeamId,
+            );
+        });
+        it("should throw badRequest exception for invalid dto", async () => {
+            mockResourcesService.createNewResource.mockRejectedValueOnce(
+                new BadRequestException(),
+            );
+            expect(
+                controller.createNewResource(requestMock, mockTeamId, {
+                    url: "https://chingu.com",
+                } as CreateResourceDto),
+            ).rejects.toThrow(BadRequestException);
+            expect(mockResourcesService.createNewResource).toHaveBeenCalledWith(
+                requestMock,
+                dtoCreateMock,
+                mockTeamId,
+            );
+        });
+        it("should throw notFound exception for invalid teamId", async () => {
+            mockResourcesService.createNewResource.mockRejectedValueOnce(
+                new NotFoundException(),
+            );
+            expect(
+                controller.createNewResource(requestMock, 9999, dtoCreateMock),
+            ).rejects.toThrow(NotFoundException);
+            expect(mockResourcesService.createNewResource).toHaveBeenCalledWith(
+                requestMock,
+                dtoCreateMock,
+                9999,
```

Comment:
extract magic numbers like this one (9999) into constants for better readability.